src/stdlib_math.fypp

```diff
@@ -1,19 +1,147 @@
 #:include "common.fypp"
 #:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
 
 module stdlib_math
     use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace, EULERS_NUMBER_DP, EULERS_NUMBER_SP
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    real(dp), parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace 
+    real(sp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
```

Comment:
Should it be `public` too?

Reply:
I'm assuming that you are referring to the `qp` version of Euler's number, correct? Tbh this might sound really silly but I'm a little afraid to mess around with `qp` because I know that on gfortran `qp` is supported but when I've tried to use `ifort` to compile, `qp` was not supported. Long story short I didn't want to break anything but I will gladly make `EULERS_NUMBER_QP` public as well